pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -131,27 +141,32 @@ func getNetworkName() (string, error) {
 
 // newGCECloud creates a new instance of GCECloud.
 func newGCECloud(config io.Reader) (*GCECloud, error) {
-	projectID, zone, err := getProjectAndZone()
-	if err != nil {
-		return nil, err
-	}
-	// TODO: if we want to use this on a machine that doesn't have the http://metadata server
-	// e.g. on a user's machine (not VM) somewhere, we need to have an alternative for
-	// instance id lookup.
-	instanceID, err := getInstanceID()
-	if err != nil {
-		return nil, err
-	}
-	externalID, err := getCurrentExternalID()
-	if err != nil {
-		return nil, err
+	var projectID, zone, instanceID, externalID, networkName, networkURL string
+	var err error
+	if onGCE() {
+		if projectID, zone, err = getProjectAndZone(); err != nil {
+			return nil, err
+		}
+		if instanceID, err = getInstanceID(); err != nil {
+			return nil, err
+		}
+		if externalID, err = getCurrentExternalID(); err != nil {
+			return nil, err
+		}
+		if networkName, err = getNetworkName(); err != nil {
+			return nil, err
+		}
+		networkURL = gceNetworkURL(projectID, networkName)
+	} else {
```

Comment:
In this else block you aren't setting most of the variables that get set when running on a GCE VM. Does that have any ramifications for callers? At minimum, we should document in the GCECloud struct which fields may be unset so that callers don't assume strings will be non-empty.